Ginger/GingerCoreNET/Database/DatabaseOperations.cs

```diff
@@ -469,65 +471,57 @@ public List<string> GetTablesList(string Keyspace = null)
+                            GingerCosmos objGingerCosmos = new GingerCosmos();
+                            Database.ConnectionString = GetConnectionString();
+                            objGingerCosmos.Db = Database;
+                            rc = objGingerCosmos.GetTableList(Keyspace);
+                        break;
+
+                        case eDBTypes.Oracle:
+                            string[] restr = new string[1];
+                            restr[0] = GetConnectedUsername();
+                            DataTable table = await oConn.GetSchemaAsync("Tables", restr);
+                            string tableName = "";
+                            foreach (DataRow row in table.Rows)
+                            {
+                                tableName = (string)row[1];
+                                rc.Add(tableName);
+                            }
+                            break;
+                        default:
+                            //Not implemented
+                            break;
                     }
-                    else
-                    {
+                        if (Database.DBType == eDBTypes.MSSQL || Database.DBType == eDBTypes.MySQL || Database.DBType == eDBTypes.MSAccess || Database.DBType == eDBTypes.DB2 || Database.DBType == eDBTypes.PostgreSQL)
+                        {
                         DataTable table = oConn.GetSchema("Tables");
                         string tableName = "";
-                        foreach (DataRow row in table.Rows)
+                        foreach (DataRow row in table.Rows) 
                         {
-                            switch (Database.DBType)
-                            {
-                                case eDBTypes.MSSQL:
-                                    tableName = (string)row[2];
-                                    break;
-                                case eDBTypes.Oracle:
-                                    tableName = (string)row[1];
-                                    break;
-                                case eDBTypes.MSAccess:
-                                    tableName = (string)row[2];
-                                    break;
-                                case eDBTypes.DB2:
-                                    tableName = (string)row[2];
-                                    break;
-                                case eDBTypes.MySQL:
-                                    tableName = (string)row[2];
-                                    break;
-                                case eDBTypes.PostgreSQL:
-                                    tableName = (string)row[2];
-                                    break;
-                                default:
-                                    //not implemented
-                                    break;
-                            }
-
-                            rc.Add(tableName);
+                                tableName = (string)row[2];
```

Comment:
directly add (string)row[2] to list without defining the string tablename. Replace it in all places

Reply:
fixed